index.js

```diff
@@ -89,59 +86,51 @@ PersonaClient.prototype.validateToken = function (token, scope, overridingScope,
         return;
     }
 
-    var _this = this,
-        cacheKey = token + ((scope) ? "@" + scope : "");
-    // if we were given a scope then append the scope to the token to create a cachekey
-
-    this.debug("Validating token: " + cacheKey);
-    this.redisClient.get("access_token:" + cacheKey, function (err, reply) {
-        if (reply === "OK") {
-            _this.debug("Token " + cacheKey + " verified by cache");
-            next(null, VALIDATED_TYPES.CACHE);
-        } else {
-            var requestPath = _this.config.persona_oauth_route + token;
-            var usingScope = scope || overridingScope || null;
+    this.debug("Validating token: " + token);
+    var _this = this;
 
-            if (usingScope != null) {
-                requestPath += "?scope=" + usingScope;
+    function verifyToken() {
+        var jwtConfig = {
+            algorithms: ["RS256"]
+        };
+
+        jwt.verify(token, _this.publicKey, jwtConfig, function onVerify(error, decodedToken) {
+            if(error) {
+                return next(ERROR_TYPES.VALIDATION_FAILURE, null);
+            }
+            _this.debug("Decoded token: " + decodedToken);
+            if (_.intersection(decodedToken.scopes, scopes).length > 0) {
+                next(null, VALIDATED_TYPE);
```

Comment:
can you prepend `return` to the `next` method to make it easier to read, less bug prone & consistent with line 99.